src/sdk/emitter/webAudioSoundEmitter.ts

```diff
@@ -8,34 +8,33 @@ import { SoundEmitter } from './soundEmitter';
  * @implements {SoundEmitter}
  */
 export class WebAudioSoundEmitter implements SoundEmitter {
-  /**
-   * Emits a sound through HTML5's AudioContext interface.
-   *
-   * @param {ArrayBuffer} sound A binary record of the sound you want to play.
-   * @returns {Promise<void>}
-   * @memberof WebAudioSoundEmitter
-   */
-  async emit(sound: ArrayBuffer): Promise<void> {
-    let context: AudioContext;
+    /**
+     * Emits a sound through HTML5's AudioContext interface.
+     *
+     * @param {ArrayBuffer} sound A binary record of the sound you want to play.
+     * @returns {Promise<void>}
+     * @memberof WebAudioSoundEmitter
+     */
+    emit(sound: ArrayBuffer) {
 
-    try {
-      context = new AudioContext();
-    } catch(e) {
-      console.error('AudioContext is not supported.');
-      
-      throw e;
-    }
+        let context: AudioContext;
 
-    const audioBuffer = await context.decodeAudioData(sound);
-    const source = context.createBufferSource();
+        try {
+            context = new (window["AudioContext"] || window["webkitAudioContext"])();
+        } catch (e) {
+            console.error('AudioContext is not supported.');
 
-    source.buffer = audioBuffer;
+            throw e;
+        }
+        const source = context.createBufferSource();
 
-    source.connect(context.destination);
-    
-    await (new Promise((resolve) => {
-      source.onended = resolve;
```

Comment:
You're breaking how the Widget works. Remember, that after a sound transmission the Widget changes its UI. It was done with `onended` callback which you've removed.
Basically, in this version we need a callback passed to this function called when the sound playing is finished.

Reply:
Add a constructor that takes the callback